File: 0940-fruit-into-baskets/0940-fruit-into-baskets.java
class Solution {
    public int totalFruit(int[] fruits) {
      HashMap<Integer,Integer> fruits_count= new HashMap<Integer,Integer>();

      
      int max_length=0;

      for(int right=0,left=0;right<fruits.length;++right){
        fruits_count.put(fruits[right], fruits_count.getOrDefault(fruits[right], 0) + 1);
        while(fruits_count.size()>2){
            fruits_count.put(fruits[left],fruits_count.get(fruits[left])-1);
            if (fruits_count.get(fruits[left])==0){
                fruits_count.remove(fruits[left],0);
            }
            ++left;

        }
        max_length=Math.max(max_length,right-left+1);

      } 
      return max_length; 
    }
}

File: 0940-fruit-into-baskets/0940-fruit-into-baskets.ts
function totalFruit(fruits: number[]): number {
    let max_length=0
    let l=0
    const fruitsCount: Map<number,number>=new Map()

    for(let r=0,l=0;r<fruits.length;++r){
        fruitsCount.set(fruits[r],(fruitsCount.get(fruits[r]) || 0)+1)
        while (fruitsCount.size>2){
            fruitsCount.set(fruits[l],(fruitsCount.get(fruits[l]) || 0)-1)
            if(fruitsCount.get(fruits[l])===0){
                fruitsCount.delete(fruits[l])
            }
            l++;
        } 
        max_length=Math.max(max_length,r-l+1)
    }
    return max_length
    
};